### PR TITLE
feat(dropdown-menu): update user login/logout flow and improve styling

### PR DESCRIPTION
**Pull request by DanielGovrin**
This PR introduces updates to the DropdownMenu component and related styles, along with API improvements for user authentication in the Wix API context provider. The main changes include:

Component Updates:

- Renamed dropdownMenuItems to menuItems for improved clarity and consistency.

- Added new login/logout functionality to the DropdownMenu, allowing dynamic display based on user authentication status.

- Wrapped the username and dropdown arrow in a new container (userMenuWrapper) for better styling and handling of hover effects.

- Introduced new SVG assets (downArrow and orangeDownArrow) for better visual feedback.

API Enhancements:

- Added initiateLogin, logout, and fetchUserAuthData methods in the Wix API context to manage user sessions.

- Improved the login/logout flow by displaying alerts for both actions during the process.

Styling:

- Updated dropdown styles in dropdown-menu.module.scss to support a more polished design and hover states for the dropdown arrows.

- Adjusted spacing and layout in the header and dropdown-menu components for consistency with the rest of the project.

### CODE REVIEW

**Comment by matangeorgi** [comment]
Please add description, maybe also include images since the changes are also visual and explain what is expected to be.
Also change the PR's title to be according to conventional commits.

File: src/_codux/fake-data/fake-wix-api-context-provider.tsx [hunk 1]
```diff
@@ -49,6 +49,16 @@ function getWixApi(settings?: Settings): WixAPI {
         getServiceAvailability: (lessonId: string) => {
             return Promise.resolve(createLessonAvailability(lessonId));
         },
+        initiateLogin: async () => {
+            alert('Initiate login');
```
Comment (TheAlmightyCrumb) [inline src/_codux/fake-data/fake-wix-api-context-provider.tsx:53]: Alert? is it needed?

File: src/_codux/fake-data/fake-wix-api-context-provider.tsx [hunk 2]
```diff
@@ -49,6 +49,16 @@ function getWixApi(settings?: Settings): WixAPI {
...
+        },
+        fetchUserAuthData: async () => {
+            return Promise.resolve({ user: { member: { profile: { nickname: 'John Doe' } } } });
```
Comment (TheAlmightyCrumb) [inline src/_codux/fake-data/fake-wix-api-context-provider.tsx:56]: Do you know if the user auth data is consistent? if so - please use types for clarity.
```suggestion
            const userAuthData = {
                user: {
                    member: {
                        profile: {
                            nickname: 'John Doe'
                        }
                    }
                }
            };
            return Promise.resolve(userAuthData);
```

File: src/_codux/fake-data/fake-wix-api-context-provider.tsx [hunk 3]
```diff
@@ -49,6 +49,16 @@ function getWixApi(settings?: Settings): WixAPI {
...
+        },
+        logout: async () => {
+            alert('Logout');
```
Comment (TheAlmightyCrumb) [inline src/_codux/fake-data/fake-wix-api-context-provider.tsx:59]: Same, will it change?

File: src/components/dropdown-menu/dropdown-menu.tsx [hunk 4]
```diff
@@ -1,27 +1,41 @@
...
 
-export const DropdownMenu = ({ dropdownMenuItems, className }: DropdownMenuProps) => {
+export const DropdownMenu = ({ menuItems, className }: DropdownMenuProps) => {
     const [isOpen, setIsOpen] = useState(false);
+    const dropDownRef = useRef<HTMLDivElement>(null);
     const wixApi = useContext(WixAPIContext);
     const [username, setUsername] = useState<string | null>(null);
 
+    const handleClickOutside = (event: MouseEvent) => {
+        if (dropDownRef.current && !dropDownRef.current.contains(event.target as Node)) {
```
Comment (TheAlmightyCrumb) [inline src/components/dropdown-menu/dropdown-menu.tsx:27]: `as Node`?

**Review by TheAlmightyCrumb — APPROVED** [review]

File: src/_codux/fake-data/fake-wix-api-context-provider.tsx (hunk 1 above)
Comment (DanielGovrin) [reply src/_codux/fake-data/fake-wix-api-context-provider.tsx:53]: removed

File: src/_codux/fake-data/fake-wix-api-context-provider.tsx (hunk 3 above)
Comment (DanielGovrin) [reply src/_codux/fake-data/fake-wix-api-context-provider.tsx:59]: removed

File: src/_codux/fake-data/fake-wix-api-context-provider.tsx (hunk 2 above)
Comment (DanielGovrin) [reply src/_codux/fake-data/fake-wix-api-context-provider.tsx:56]: added type to user auth data

**Pull request merged**